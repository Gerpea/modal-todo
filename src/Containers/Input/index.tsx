import React from 'react';
import clsx from 'clsx';

import style from './styles.module.scss';

export interface Props {
  className?: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  value?: string;
  placeholder?: string;
}

function Input({ className, value, placeholder, onChange }: Props): React.ReactElement {
  const styles = clsx(className, style.Input);

  return <input value={value} className={styles} onChange={onChange} placeholder={placeholder} />;
}

export default Input;
