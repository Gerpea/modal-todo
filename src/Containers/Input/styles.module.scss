@import '../../variables';

.Input {
  position: relative;

  outline: none;
  border: none;
  background-color: transparent;

  width: 100%;
  padding: 12px 0;

  border-bottom: 1px solid transparentize($c-blue, 0.5);
  color: $c-black;

  font-size: 1rem;
  line-height: 1.125;
}

.Input::placeholder {
  color: transparentize($c-blue, 0.5);
}

.Input:focus {
  border-color: $c-blue;
}
