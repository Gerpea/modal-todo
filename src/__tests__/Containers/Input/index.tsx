import React from 'react';
import renderer from 'react-test-renderer';

import Input from 'Containers/Input';

describe('Input test', () => {
  it('Input renders', () => {
    const tree = renderer.create(<Input onChange={() => void {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
