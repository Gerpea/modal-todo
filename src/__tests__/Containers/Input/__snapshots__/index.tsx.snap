// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test Input renders 1`] = `
<input
  className="Input"
  onChange={[Function]}
/>
`;
